trace/tracestate/tracestate_test.go

```diff
@@ -0,0 +1,320 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracestate
+
+import (
+	"fmt"
+	"testing"
+)
+
+func checkFront(t *testing.T, tracestate *Tracestate, wantKey, testname string) {
+	gotKey := tracestate.entries[0].Key
+	if gotKey != wantKey {
+		t.Errorf("test:%s: first entry in the list: got %q want %q", testname, gotKey, wantKey)
+	}
+}
+
+func checkBack(t *testing.T, tracestate *Tracestate, wantKey, testname string) {
+	gotKey := tracestate.entries[len(tracestate.entries)-1].Key
+	if gotKey != wantKey {
+		t.Errorf("test:%s: last entry in the list: got %q want %q", testname, gotKey, wantKey)
+	}
+}
+
+func checkSize(t *testing.T, tracestate *Tracestate, wantSize int, testname string) {
+	gotSize := len(tracestate.entries)
+	if gotSize != wantSize {
```

Comment:
You can write these inline.

if got, want := len(tracestate.entries), wantSize  {}